bootstrap/bootstrap.go

```diff
@@ -639,45 +638,62 @@ func setupStorage(
 	}, nil
 }
 
-// Run will run complete bootstrap of the EVM gateway with all the engines.
-// Run is a blocking call, but it does signal readiness of the service
-// through a channel provided as an argument.
-func Run(ctx context.Context, cfg config.Config, ready component.ReadyFunc) error {
-	boot, err := New(cfg)
-	if err != nil {
-		return err
-	}
-
+func (b *Bootstrap) Run(
+	ctx context.Context,
+	cfg config.Config,
+	ready component.ReadyFunc,
+) error {
 	// Start the API Server first, to avoid any races with incoming
 	// EVM events, that might affect the starting state.
-	if err := boot.StartAPIServer(ctx); err != nil {
+	if err := b.StartAPIServer(ctx); err != nil {
 		return fmt.Errorf("failed to start API server: %w", err)
 	}
 
-	if err := boot.StartEventIngestion(ctx); err != nil {
+	if err := b.StartEventIngestion(ctx); err != nil {
 		return fmt.Errorf("failed to start event ingestion engine: %w", err)
 	}
 
-	if err := boot.StartMetricsServer(ctx); err != nil {
+	if err := b.StartMetricsServer(ctx); err != nil {
 		return fmt.Errorf("failed to start metrics server: %w", err)
 	}
 
-	if err := boot.StartProfilerServer(ctx); err != nil {
+	if err := b.StartProfilerServer(ctx); err != nil {
 		return fmt.Errorf("failed to start profiler server: %w", err)
 	}
 
 	// mark ready
 	ready()
 
+	return nil
+}
+
+func (b *Bootstrap) Stop() {
+	b.logger.Warn().Msg("bootstrap received context cancellation, stopping services")
```

Comment:
I don't think we need to warn about a graceful shutdown
```suggestion
	b.logger.Info().Msg("bootstrap received context cancellation, stopping services")
```